docs/source/refs/reference_architecture/index.rst

```diff
@@ -0,0 +1,392 @@
+In running your RL task, you might want to change the behavior of your environment without
+changing the environment itself. For example, you might want to create functions to modify
+observations or rewards, record videos, or enforce time limits. Isaac Lab utilizes the API
+available in the `gymnasium.Wrapper <https://gymnasium.farama.org/api/wrappers/#gymnasium.Wrapper>`__ class to create interfaces to the simulated environments.
+
+Some wrappers include:
+
+* `Video Wrappers <https://isaac-sim.github.io/IsaacLab/main/source/how-to/wrap_rl_env.html#wrapper-for-recording-videos>`__
+* `RL Libraries Wrappers <https://isaac-sim.github.io/IsaacLab/main/source/how-to/wrap_rl_env.html#wrapper-for-learning-frameworks>`__
+
+Most RL libraries expect their own variation of an environment interface. This means the
+data types needed by each library differs. Isaac Lab provides its own wrappers to convert
+the environment into the expected interface by the RL library a user wants to use. These are
+specified in the `Isaac Lab utils wrapper module <https://isaac-sim.github.io/IsaacLab/main/source/api/lab_tasks/omni.isaac.lab_tasks.utils.wrappers.html#module-omni.isaac.lab_tasks.utils.wrappers>`__.
+
+See the `full list <https://gymnasium.farama.org/api/wrappers/#gymnasium.Wrapper>`__ of other wrappers APIs.. For more information on how these wrappers work,
+please refer to the `Wrapping environments <https://isaac-sim.github.io/IsaacLab/main/source/how-to/wrap_rl_env.html#how-to-env-wrappers>`__ documentation.
+
+**Adding your own wrappers**
+
+You can define your own wrappers by adding them to the Isaac Lab utils wrapper module. More information is available `on the GitHub page for wrapping environments <https://isaac-sim.github.io/IsaacLab/main/source/how-to/wrap_rl_env.html#adding-new-wrappers>`__.
+
+**Component 6 - Run Training**
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Finally, the last step is to run the training of the RL agent. Isaac Lab provides scripts which utilizes four popular RL libraries for training the models (GPU-based training):
+
+* `StableBaselines3 <https://stable-baselines3.readthedocs.io/en/master/>`__
+* `RSL-RL <https://github.com/leggedrobotics/rsl_rl>`__
+* `RL-Games <https://github.com/Denys88/rl_games>`__
+* `SKRL <https://skrl.readthedocs.io/en/latest/>`__
+
+
+.. note::
+
+  Isaac Lab does not provide the implementation of these RL libraries. They are already implemented by different authors. We provide the environment and the training framework that takes place with the popular RLlibraries.
+
+
+
+If you want to integrate a different version of the provided algorithms or your learning library, you can follow
+`these instructions <https://isaac-sim.github.io/IsaacLab/main/source/how-to/add_own_library.html>`__.
+
+.. note::
+
+  See the  `minimum system requirements <https://isaac-sim.github.io/IsaacLab/main/source/setup/installation/index.html>`__ for training using Isaac Lab.
+
+
+**Single GPU Training**
```

Comment:
I think this whole section about single GPU, multi GPU, and multi node go into too much detail. we have a dedicated page to explain multi-GPU. maybe we should just quickly mention the supported workflows here without going into too much depth.

Reply:
I have removed the information on the multi-GPU section. The single GPU is mainly explaining how data flows